test/python/algorithms/evolvers/variational/calculators/test_evolution_grad_calculator.py

```diff
@@ -0,0 +1,215 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test evolution gradient calculator."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from ddt import data, unpack, ddt
+import numpy as np
+
+from qiskit.algorithms.evolvers.variational.calculators.evolution_grad_calculator import (
+    calculate,
+)
+from qiskit.circuit.library import EfficientSU2
+from qiskit.opflow import SummedOp, X, Y, I, Z
+
+
+@ddt
+class TestEvolutionGradCalculator(QiskitAlgorithmsTestCase):
+    """Test evolution gradient calculator."""
+
+    # checked, correct
+    def test_calculate(self):
+        """Test calculating evolution gradient."""
+        observable = SummedOp(
+            [
+                0.2252 * (I ^ I),
+                0.5716 * (Z ^ Z),
+                0.3435 * (I ^ Z),
+                -0.4347 * (Z ^ I),
+                0.091 * (Y ^ Y),
+                0.091 * (X ^ X),
+            ]
+        ).reduce()
+
+        d = 2
+        ansatz = EfficientSU2(observable.num_qubits, reps=d)
+
+        # Define a set of initial parameters
+        parameters = ansatz.ordered_parameters
+        grad_method = "lin_comb"
+        evolution_grad = calculate(observable, ansatz, parameters, grad_method)
```

Comment:
We'd like to deprecate `ordered_parameters` since parameters are now anyway sorted by name and that attribute is not necessary anymore,  so better use `parameters` directly 
```suggestion
        grad_method = "lin_comb"
        evolution_grad = calculate(observable, ansatz, ansatz.parameters, grad_method)
```